tilings/strategies/fusion.py

```diff
@@ -1,28 +1,147 @@
-from typing import Iterator, Optional, Tuple
+from collections import defaultdict
+from itertools import product
+from typing import Callable, Dict, Iterable, Iterator, List, Optional, Set, Tuple
 
 from sympy import Eq, Function
 
-from comb_spec_searcher import (
-    CombinatorialObject,
-    Constructor,
-    Strategy,
-    StrategyFactory,
-)
+from comb_spec_searcher import Constructor, Strategy, StrategyFactory
+from comb_spec_searcher.exception import StrategyDoesNotApply
 from comb_spec_searcher.strategies import Rule
-from comb_spec_searcher.strategies.constructor import (
-    RelianceProfile,
-    SubGens,
-    SubRecs,
-    SubSamplers,
-)
+from comb_spec_searcher.strategies.constructor import RelianceProfile, SubRecs
 from tilings import GriddedPerm, Tiling
 from tilings.algorithms import ComponentFusion, Fusion
+from tilings.assumptions import TrackingAssumption
 
 __all__ = ["FusionStrategy", "ComponentFusionStrategy"]
 
+SubGens = Tuple[Callable[..., Iterator[GriddedPerm]], ...]
+SubRec = Callable[..., int]
+SubSamplers = Tuple[Callable[..., GriddedPerm], ...]
 
-class FusionConstructor(Constructor):
-    def is_equivalence(self) -> bool:
+
+class FusionConstructor(Constructor[Tiling, GriddedPerm]):
+    """
+    The fusion constructor. It will multiply by (fuse_paramater + 1), and
+    otherwise pass on the variables.
+
+    The parameters given should be a dictionary. Each child variable should
+    point to some parent variable with the exception of the fuse variable which
+    will not point if it was just added. If [ A | A ] fuses to [ A ] then we
+    assume any one sided variable maps to the [ A ].
+    """
+
+    def __init__(
+        self,
+        fuse_parameter: str,
+        extra_parameters: Dict[str, str],
+        left_sided_parameters=Iterable[str],
+        right_sided_parameters=Iterable[str],
+        both_sided_parameters=Iterable[str],
+    ):
```

Comment:
Can we add a docstring that describe what are the arguments

Reply:
Will do.